api/v1alpha1/helper.go

```diff
@@ -0,0 +1,30 @@
+package v1alpha1
+
+import "encoding/json"
+
+// CanaryMessage record each changes of rollout-id in Rollout.Status.CanaryStatus.Message field.
+// PaaS can use this message to judge whether pods would be upgraded when users publishing resources.
+// For example, users may just modify Service instead of Workload, PaaS can use this message to distinguish such scene.
+type CanaryMessage struct {
+	// RolloutID indicate the corresponding to rollout-id recorded in this message
```

Comment:
it is not necessary to encode rollout id in the rollout status, since there is already an ObservedRolloutID field in the status. and the message filed should be design for human reading only, it is not appropriate to encode structured data in the message.